extensions/typescript-language-features/src/tsServer/versionManager.ts

```diff
@@ -101,16 +121,19 @@ export class TypeScriptVersionManager extends Disposable {
 	private getLocalPickItems(): QuickPickItem[] {
 		return this.versionProvider.localVersions.map(version => {
 			return {
-				label: (this.useWorkspaceTsdkSetting && this.currentVersion.eq(version)
+				label: (this.useWorkspaceTsdkSetting && this.isWorkspaceTrusted && this.currentVersion.eq(version)
 					? '• '
 					: '') + localize('useWorkspaceVersionOption', "Use Workspace Version"),
 				description: version.displayName,
 				detail: version.pathLabel,
 				run: async () => {
-					await this.workspaceState.update(useWorkspaceTsdkStorageKey, true);
-					const tsConfig = vscode.workspace.getConfiguration('typescript');
-					await tsConfig.update('tsdk', version.pathLabel, false);
-					this.updateActiveVersion(version);
+					const trustState = await vscode.workspace.requireWorkspaceTrust();
+					if (trustState === vscode.WorkspaceTrustState.Trusted) {
```

Comment:
If the user denies trust, is there some way to inform the user why their local TS version is not being used?

Reply:
Later on when implemented in core, when they are asked about trust, they are informed that this will be the impact. They can also see this in the workspace trust settings UI. Technically nothing is stopping TS from showing some extra info, but I think we should avoid that as a pattern.